install/installer/pkg/components/registry-credential/cronjob.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package registry_credential
+
+import (
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/pointer"
+
+	"github.com/gitpod-io/gitpod/installer/pkg/common"
+)
+
+func cronjob(ctx *common.RenderContext) ([]runtime.Object, error) {
+	objectMeta := metav1.ObjectMeta{
+		Name:      Component,
+		Namespace: ctx.Namespace,
+		Labels:    common.DefaultLabels(Component),
+	}
+
+	return []runtime.Object{
+		&batchv1.CronJob{
+			TypeMeta:   common.TypeMetaBatchCronJob,
+			ObjectMeta: objectMeta,
+			Spec: batchv1.CronJobSpec{
+				Schedule:                   CronSchedule,
+				SuccessfulJobsHistoryLimit: pointer.Int32(1),
+				FailedJobsHistoryLimit:     pointer.Int32(10),
+				ConcurrencyPolicy:          batchv1.ReplaceConcurrent,
+				JobTemplate: batchv1.JobTemplateSpec{
+					ObjectMeta: objectMeta,
+					Spec: batchv1.JobSpec{
+						BackoffLimit: pointer.Int32(10),
```

Comment:
Looks like we're using the built-in backoff feature for cronjobs, nice! How long will it take to do the 10 retries? Is there time cap we could hit, before we get to the 10th attempt? I ask because I don't know how the cap is calculated. For example, with ws-manager, we do like [this](https://github.com/gitpod-io/gitpod/blob/ae4f39db43423e83edbeddaba3fe9f4b9d0b9f52/components/ws-manager/pkg/manager/manager.go#L1601-L1607).

Reply:
Check this https://github.com/gitpod-io/gitpod/pull/15313#issuecomment-1352442617